StratisMasternodeDashboard-master/Services/ApiRequester.cs

```diff
@@ -58,5 +61,57 @@ public async Task<ApiResponse> PostRequestAsync(string endpoint, string path, ob
                 Content = JsonConvert.DeserializeObject(restResponse.Content)
             };
         }
+
+        #region Kick
+        protected async Task<List<FederationMember>> GetFederationMembers(string endpoint)
+        {
+            List<FederationMember> federationMembers = new List<FederationMember>();
+            try
+            {
+                ApiResponse response = await GetRequestAsync(endpoint, "/api/Federation/members");
+                federationMembers = JsonConvert.DeserializeObject<List<FederationMember>>(response.Content.ToString());
+                return federationMembers;
+            }
+            catch (Exception ex)
+            {
+                this.logger.LogError(ex, "Failed to get Federation Members");
+            }
+            return federationMembers;
+        }
+
+        public async Task<IDGBKickModel> CheckAndKickFeberationMember(string endpoint, string pubKey)
+        {
+            bool isKeyAvailable = false;
+            bool isSuccess = false;
+            string message=null;
+            IDGBKickModel iDBGKick = new IDGBKickModel();
+            try
+            {
+                var members = await GetFederationMembers(endpoint);
```

Comment:
Satya, whilst its not incorrect to check if the federation member that you are trying to kick exists or not, it is already handled in the FullNode's api method.

You can remove this logic and literally only call the endpoint and return any messages if need be.